### PR TITLE
[batch] Configurable worker disk type

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #9285

### CODE REVIEW

**Comment by jigold** [comment]
We decided to ignore this for now.

**Comment by jigold** [comment]
This is passing the tests. I didn't double check the billing numbers in the database look correct. Can do so if you think it's prudent.

File: batch/batch/worker_config.py [hunk 1]
```diff
@@ -98,6 +98,11 @@ def __init__(self, config):
         self.cores = instance['cores']
         self.preemptible = instance['preemptible']
 
+        data_disk = [d for d in self.disks if not d['boot']][0]
+        assert data_disk
+        self.local_ssd_data_disk = (data_disk['type'] == 'local-ssd')
+        self.data_disk_size_gb = data_disk['size']
```
Comment (cseed) [inline batch/batch/worker_config.py:104]: Can we change this to asserting there are two disks and the first one is the boot disk, and just take the second one?  This code needs to change if there are >2 disks.

File: batch/batch/worker/worker.py [hunk 2]
```diff
@@ -677,21 +677,15 @@ def __init__(self, batch_id, user, gsa_key, job_spec, format_version):
 
         req_cpu_in_mcpu = parse_cpu_in_mcpu(job_spec['resources']['cpu'])
         req_memory_in_bytes = parse_memory_in_bytes(job_spec['resources']['memory'])
-        if 'storage' in job_spec['resources']:
-            req_storage_in_bytes = parse_memory_in_bytes(job_spec['resources']['storage'])
```
Comment (cseed) [inline batch/batch/worker/worker.py:681]: Just want to confirm this was due to Ben's bug and new job specs no longer have a `storage` key?

Comment (jigold) [reply batch/batch/worker/worker.py:681]: The opposite. All job_specs should have a "resources" dictionary with "storage" in it now.

**Comment by jigold** [comment]
I addressed the comment. The only thing to note is the default storage is set to 10 Gi I believe and I'm subtracting 25 Gi from the persistent disk size when figuring out how many cores the job requires. @konradjk

File: batch/batch/worker/worker.py (hunk 2 above)
Comment (cseed) [reply batch/batch/worker/worker.py:681]: Got it, thanks!

**Pull request merged**